forward/utils.go

```diff
@@ -17,9 +17,27 @@ func cloneURL(i *url.URL) *url.URL {
 // copyHeaders copies http headers from source to destination.  It does not
 // overide, but adds multiple headers
 func copyHeaders(dst, src http.Header) {
+	var extraHopByHopHeaders []string
 	for k, vv := range src {
-		for _, v := range vv {
-			dst.Add(k, v)
+		switch k {
```

Comment:
A `switch` statement might be more efficient than an array lookup which is O(n). A map lookup is O(1) but seems a bit unusual. We also have special case on "Connection" header. `switch` here is straightforward to me but improvements are welcome.